drivers/timer/ti_j721e_dmtimer.c

```diff
@@ -0,0 +1,75 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_j721e_dmtimer
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/drivers/timer/ti_j721e_dmtimer.h>
+
+#define CYC_PER_TICK  ((uint32_t)(sys_clock_hw_cycles_per_sec() \
+				/ CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+
+static struct k_spinlock lock;
+
+void ti_j721e_dmtimer_isr(void *data)
+{
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	sys_write32(BIT(TI_J721E_DM_TIMER_IRQENABLE_SET_OVF_EN_FLAG_SHIFT),
+							TI_J721E_DM_TIMER_IRQSTATUS);
+	sys_write32(0xffffffff - CYC_PER_TICK, TI_J721E_DM_TIMER_TLDR);
+	sys_write32(0xffffffff - CYC_PER_TICK, TI_J721E_DM_TIMER_TCRR);
+	sys_write32(BIT(TI_J721E_DM_TIMER_TCLR_ST_SHIFT), TI_J721E_DM_TIMER_TCLR);
+
+	k_spin_unlock(&lock, key);
+
+	sys_clock_announce(1);
+}
+
+uint32_t sys_clock_cycle_get_32(void)
+{
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	uint32_t tcrr_count = sys_read32(TI_J721E_DM_TIMER_TCRR);
+
+	k_spin_unlock(&lock, key);
+
+	return (tcrr_count - (0xffffffff - CYC_PER_TICK));
+}
+
+unsigned int sys_clock_elapsed(void)
+{
+	return 0;
+}
```

Comment:
@slpp95prashanth I don't think this can be unimplemented.
Also missing `sys_clock_set_timeout()` implementation?

Reply:
@vaishnavachath 
sys_clock_elapsed returns zero if TICKLESS_CAPABLE is disabled.

sys_clock_set_timeout() is required for TICKLESS_CAPABLE option. I tested the threads and time features seems to work. I thought of adding TICKLESS_CAPABLE once initial support is merged. Shall I add it after the soc, board, device tree, interrupt controller and timers PR are merged?